schedules/app/controllers/search_controller.rb

```diff
@@ -33,8 +33,10 @@ def index
 
     /[0-9]{5}/.match(params[:query]) do |m|
       course = CourseSection.latest_by_crn(m[0])
-      url = course_url(course.course) + "#section-" + course.id.to_s
-      redirect_to(url)
+      if not course.nil?
```

Comment:
Minor code clarity nitpick: Ruby has an `unless` keyword that is equivalent to `if not`. Try using that here and moving the redirect into the unless statement!